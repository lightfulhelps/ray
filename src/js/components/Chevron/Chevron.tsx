import React from 'react';
import Icon from '../Icon/Icon';
import { IconNameType } from '../Icon/icons';
import Button from '../Button/Button';

type ChevronDirection = 'left' | 'right' | 'up' | 'down';

type Props = {
  direction?: ChevronDirection;
  onClick: () => void;
  className?: string;
  [key: string]: any;
};

const dirToIconName = (direction: ChevronDirection): IconNameType => {
  switch (direction) {
    case 'left':
      return 'chevronLeft';
    case 'right':
      return 'chevronRight';
    case 'up':
      return 'chevronUp';
    case 'down':
      return 'chevronDown';
    default:
      return 'chevronLeft';
  }
};

const Chevron: React.FC<Props> = ({ direction = 'left', onClick, className, ...other }: Props) => {
  return (
    <Button
      onClick={onClick}
      size="sm"
      className={`chevron-button ${className || ''}`}
      theme="gray-300"
      {...other}
    >
      <Icon name={dirToIconName(direction)} color="gray-600" />
    </Button>
  );
};

export default Chevron;
