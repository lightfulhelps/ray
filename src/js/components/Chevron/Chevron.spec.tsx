import React from 'react';
import { shallow } from 'enzyme';
import Chevron from './Chevron';

describe('Chevron', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Chevron onClick={() => console.log('click')} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have the correct class', () => {
    const wrapper = shallow(<Chevron className="foo" onClick={() => console.log('click')} />);
    expect(wrapper.hasClass('chevron-button foo')).toBe(true);
  });

  it('should have the defaut left chevron direction icon', () => {
    const wrapper = shallow(<Chevron onClick={() => console.log('click')} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle the direction correctly and change the icon', () => {
    const wrapper = shallow(<Chevron direction="right" onClick={() => console.log('click')} />);
    expect(wrapper).toMatchSnapshot();
  });
});
