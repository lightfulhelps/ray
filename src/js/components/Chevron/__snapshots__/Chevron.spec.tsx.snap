// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chevron should handle the direction correctly and change the icon 1`] = `
<Button
  className="chevron-button "
  onClick={[Function]}
  size="sm"
  theme="gray-300"
>
  <Icon
    color="gray-600"
    name="chevronRight"
  />
</Button>
`;

exports[`Chevron should have the defaut left chevron direction icon 1`] = `
<Button
  className="chevron-button "
  onClick={[Function]}
  size="sm"
  theme="gray-300"
>
  <Icon
    color="gray-600"
    name="chevronLeft"
  />
</Button>
`;

exports[`Chevron should render correctly 1`] = `
<Button
  className="chevron-button "
  onClick={[Function]}
  size="sm"
  theme="gray-300"
>
  <Icon
    color="gray-600"
    name="chevronLeft"
  />
</Button>
`;
